### PR TITLE
Redesign UserGroups creation process 

### PR DESCRIPTION
**Pull request by hidesminimally**
UserGroups are now created only when the user _joins_ a group. They no longer have the `is_member` attribute, only `is_leader` attribute. The existence of a user_group implies the user is a member to that group. 
- Add auto-add group creator as leader, and creates user_group for them 
- Finish MemberListing method 

@nikitr

### CODE REVIEW

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 .landing {
   text-align: center;
-  background: url("bg_main.jpg") transparent no-repeat;
+  background: image-url("bg_main.jpg") transparent no-repeat;
```
Comment (hidesminimally) [inline app/assets/stylesheets/static_pages.scss:22]: @andylouisqin @shannonchu we can remove this in the next PR if needed

File: app/controllers/groups_controller.rb [hunk 2]
```diff
@@ -13,20 +13,21 @@ def show
...
     end
   end
 
+
+  # When User creates group, create a UserGroup with is_leader: true
   def create
     @group = Group.new(group_params)
     respond_to do |format|
       if @group.save
+        user_group = UserGroup.create(user: current_user, group: @group, is_leader: true)
```
Comment (hidesminimally) [inline app/controllers/groups_controller.rb]: @nikitr We're not creating a UserGroup for everyone anymore

File: app/controllers/groups_controller.rb [hunk 3]
```diff
@@ -34,13 +35,14 @@ def create
     end
   end
 
+
   def member_listing
     @group = Group.friendly.find(params[:id])
-    @members = @group.users
+    user_ids = @group.user_groups.pluck(:user_id)
```
Comment (hidesminimally) [inline app/controllers/groups_controller.rb]: @nikitr We're finding the user_groups, which indicate membership.

File: app/models/user.rb [hunk 4]
```diff
@@ -6,7 +6,11 @@ class User < ActiveRecord::Base
   has_many :neighborhoods, through: :user_neighborhoodss
 
   has_many :user_groups
-  has_many :groups, through: :user_groups
+  has_many :groups, through: :user_groups do
```
Comment (hidesminimally) [inline app/models/user.rb:9]: Using this scope, we're able to do `current_user.groups.leader_of` and it gives us the list of groups that current user is leading @andylouisqin @HacksForHugs @nikitr

File: app/models/group.rb [hunk 5]
```diff
@@ -8,7 +7,11 @@ class Group < ActiveRecord::Base
   belongs_to :neighborhood
 
   has_many :user_groups
-  has_many :users, through: :user_groups
+  has_many :users, through: :user_groups do
```
Comment (hidesminimally) [inline app/models/group.rb:10]: Using this scope, we're able to do `current_group.users.leaders` to query all the leaders of that group.

File: app/controllers/groups_controller.rb [hunk 6]
```diff
@@ -13,20 +13,20 @@ def show
...
 
   def new
-    @group = Group.new
-    @neighborhoods = Neighborhood.all.map{|u| [ u.name, u.id ] }
-
     if current_user
-     @user = current_user
+      @group = Group.new
+      @neighborhoods = Neighborhood.all.map{|u| [ u.name, u.id ] }
     else
```
Comment (aykamko) [inline app/controllers/groups_controller.rb:19]: Seems like indentation is off here

File: app/controllers/groups_controller.rb (hunk 2 above)
Comment (aykamko) [inline app/controllers/groups_controller.rb]: I don't think you should be creating `UserGroups` manually. There should be a method like:

``` ruby
@group.add_user(current_user, make_leader=true)
```

The UserGroup stuff would happen behind the scenes

File: app/controllers/groups_controller.rb [hunk 7]
```diff
@@ -36,11 +36,11 @@ def create
 
   def member_listing
     @group = Group.friendly.find(params[:id])
-    @members = @group.users
+    user_ids = @group.user_groups.pluck(:user_id)
+    @members = User.where(id: user_ids)
```
Comment (aykamko) [inline app/controllers/groups_controller.rb]: Should also abstract the above two lines into the `Group` model.

``` ruby
@members = @group.members()
```

All the business logic dealing with `UserGroup` should be contained in the `Group` model

File: app/models/group.rb (hunk 5 above)
Comment (aykamko) [inline app/models/group.rb:10]: nice, these are the kind of abstractions you should use

**Comment by aykamko** [comment]
logic looks almost sound now! fix my nits and ping me on slack. :+1:

**Comment by aykamko** [comment]
looks good!

**Pull request merged**